### PR TITLE
MRG FIX: Remove deprecated code

### PR DESCRIPTION
**Pull request by Eric89GXL**
+110 / -1307 isn't bad :)

I expect a few Travis errors because I haven't run tests, and Travis will get them done faster than my laptop.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
@dengemann ready for review/merge from my end :) Should probably be the first thing we go for for 0.10 so people don't mess with deprecated stuff

File: mne/source_estimate.py [hunk 1]
```diff
@@ -1352,10 +1278,9 @@ def center_of_mass(self, subject=None, hemi=None, restrict_vertices=False,
 
     def plot(self, subject=None, surface='inflated', hemi='lh',
              colormap='auto', time_label='time=%0.2f ms',
-             smoothing_steps=10, fmin=None, fmid=None, fmax=None,
-             transparent=None, alpha=1.0, time_viewer=False,
-             config_opts={}, subjects_dir=None, figure=None,
-             views='lat', colorbar=True, clim=None):
+             smoothing_steps=10, transparent=None, alpha=1.0,
+             time_viewer=False, config_opts={}, subjects_dir=None, figure=None,
```
Comment (dengemann) [inline mne/source_estimate.py]: @Eric89GXL this is opens the door to state-relate bug when the function is called multiple times, the the dict will still be the same. A no-go. We should put it to None and handle it inside.

Comment (dengemann) [inline mne/source_estimate.py]: A pitty this slipped through the review.

File: mne/viz/_3d.py [hunk 2]
```diff
@@ -431,10 +427,10 @@ def _limits_to_control_points(clim, stc_data, colormap):
 
 def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
                           colormap='auto', time_label='time=%0.2f ms',
-                          smoothing_steps=10, fmin=None, fmid=None, fmax=None,
-                          transparent=None, alpha=1.0, time_viewer=False,
-                          config_opts={}, subjects_dir=None, figure=None,
-                          views='lat', colorbar=True, clim=None):
+                          smoothing_steps=10, transparent=None, alpha=1.0,
+                          time_viewer=False, config_opts={}, subjects_dir=None,
```
Comment (dengemann) [inline mne/viz/_3d.py]: again mutable defaults in function ...

**Comment by dengemann** [comment]
@Eric89GXL looks good to me. But the mutable defaults we have in 0.9 code can be considered bug makers.

**Comment by Eric89GXL** [comment]
I can fix it in this PR. I think backporting a fix will be a bit overkill. It's been that way for a long time, and I don't think we or PySurfer actually modifies the dict.

**Comment by dengemann** [comment]
>  I think backporting a fix will be a bit overkill. It's been that way for a long time, and I don't think we or PySurfer actually modifies the dict.

Fine by me but please fix it here. This is why we started with this defaults stuff in the first place which we recently polished.

**Comment by Eric89GXL** [comment]
@dengemann all better, feel free to merge if you're happy

**Comment by dengemann** [comment]
Thanks @Eric89GXL !!!

**Pull request merged**